Gems/Terrain/Code/Source/Components/TerrainWorldRendererComponent.cpp

```diff
@@ -85,15 +99,65 @@ namespace Terrain
+                    ->DataElement(AZ::Edit::UIHandlers::Slider, &ClipmapConfiguration::m_detailClipmapScaleBase, "Detail clipmap scale base", "The scale base between two adjacent clipmap layers.")
+                        ->Attribute(AZ::Edit::Attributes::Min, 2.0f)
+                        ->Attribute(AZ::Edit::Attributes::SoftMin, 2.0f)
+                        ->Attribute(AZ::Edit::Attributes::Max, 4.0f)
+                        ->Attribute(AZ::Edit::Attributes::SoftMax, 4.0f)
+                    ->DataElement(AZ::Edit::UIHandlers::Slider, &ClipmapConfiguration::m_macroClipmapMarginSize, "Macro clipmap margin size", "The margin of the clipmap where the data won't be used.")
+                        ->Attribute(AZ::Edit::Attributes::Min, 1u)
+                        ->Attribute(AZ::Edit::Attributes::SoftMin, 1u)
+                        ->Attribute(AZ::Edit::Attributes::Max, 8u)
+                        ->Attribute(AZ::Edit::Attributes::SoftMax, 8u)
+                    ->DataElement(AZ::Edit::UIHandlers::Slider, &ClipmapConfiguration::m_detailClipmapMarginSize, "Detail clipmap margin size", "The margin of the clipmap where the data won't be used.")
+                        ->Attribute(AZ::Edit::Attributes::Min, 1u)
+                        ->Attribute(AZ::Edit::Attributes::SoftMin, 1u)
+                        ->Attribute(AZ::Edit::Attributes::Max, 8u)
+                        ->Attribute(AZ::Edit::Attributes::SoftMax, 8u)
+                    // Note: m_extendedClipmapMarginSize, m_clipmapBlendSize won't be exposed because algorithm may change and we may not need them.
+                    ;
+
                 editContext->Class<TerrainWorldRendererConfig>("Terrain World Renderer Component", "Enables terrain rendering")
                     ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
                         ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZStd::vector<AZ::Crc32>({ AZ_CRC_CE("Level") }))
                         ->Attribute(AZ::Edit::Attributes::Visibility, AZ::Edit::PropertyVisibility::ShowChildrenOnly)
                         ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
                     ->DataElement(AZ::Edit::UIHandlers::Default, &TerrainWorldRendererConfig::m_meshConfig, "Mesh configuration", "")
-                    ->Attribute(AZ::Edit::Attributes::Visibility, AZ::Edit::PropertyVisibility::ShowChildrenOnly)
+                        ->Attribute(AZ::Edit::Attributes::Visibility, AZ::Edit::PropertyVisibility::ShowChildrenOnly)
                     ->DataElement(AZ::Edit::UIHandlers::Default, &TerrainWorldRendererConfig::m_detailMaterialConfig, "Detail material configuration", "")
                         ->Attribute(AZ::Edit::Attributes::Visibility, AZ::Edit::PropertyVisibility::ShowChildrenOnly)
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &TerrainWorldRendererConfig::m_clipmapConfig, "Clipmap configuration", "")
                         ;
```

Comment:
Would it be possible to also expose a toggle to turn clipmaps completely on and off?

Reply:
I'll work on this next.